runtime/metricsview/executor.go

```diff
@@ -59,6 +60,7 @@ func NewExecutor(ctx context.Context, rt *runtime.Runtime, instanceID string, mv
 // Close releases the resources held by the Executor.
 func (e *Executor) Close() {
 	e.olapRelease()
+	_ = os.RemoveAll(e.tempDir)
```

Comment:
Unfortunately this is not safe because the executor might be closed before the caller is done with the export file. The `Export` function states `The caller is responsible for cleaning up the file.`

We probably need to handle deletion of the file in the individual error cases.

Reply:
I had checked that in both cases i.e. downloads and pivot the file is consumed before executor is closed. 

The problem is that we also need to clear the directory path (`metrics_export*`) and not just the file otherwise we are left with many empty paths in `temp_dir`. Given the temp_dir only gets cleared on server restarts this can add up to a lot of empty directories which is not ideal. 
I can see following solutions to handle such situations:
1. The solution presented here. We can change the comment but I agree it is not ideal.
2. When the creator does not delete the file then the files should be added directly into the `temp_dir` or a unique directory across all calls and not a random child directory.
3. The caller to `executor.Export` passes a path. 

Please let me know what do you think. I think we should not leave empty directories. We can go with option 2.

Reply:
By option 2, you basically mean outputting to `<temp dir>/metrics_export<rand>.<format>` instead of `<temp dir>/metrics_export<rand>/name.<format>`? I think that sounds good.